sdk/core/azure-core/src/main/java/com/azure/core/implementation/RestProxy.java

```diff
@@ -154,6 +164,78 @@ public Object invoke(Object proxy, final Method method, Object[] args) {
         }
     }
 
+    private Flux<ByteBuffer> validateLength(final HttpRequest request, final Flux<ByteBuffer> bbFlux) {
+        if (bbFlux == null) {
+            return Flux.empty();
+        }
+        return Flux.defer(new Supplier<Publisher<ByteBuffer>>() {
+            @Override
+            public Publisher<ByteBuffer> get() {
+                Long expectedLength = Long.valueOf(request.headers().value("Content-Length"));
+                List<Integer> bufferLengthList = new ArrayList<>();
+                return bbFlux.doOnEach(s -> {
+                    if (s.isOnNext()) {
+                        Long currentTotalLength = Long.valueOf(
+                            bufferLengthList.stream().reduce(Integer::sum).orElse(0)) + s.get().remaining();
```

Comment:
Is it possible for this or the other reduce method running into an overflow issue?

Reply:
Good call.
Make the reduce take in long.